src/frontend/js/pages/TeacherDashboardContractsLayout/components/BulkDownloadContractButton/index.timer.spec.tsx

```diff
@@ -0,0 +1,143 @@
+import { faker } from '@faker-js/faker';
+import { render, screen, waitFor } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import { PropsWithChildren } from 'react';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { RichieContextFactory as mockRichieContextFactory } from 'utils/test/factories/richie';
+import JoanieApiProvider from 'contexts/JoanieApiContext';
+import { createTestQueryClient } from 'utils/test/createTestQueryClient';
+
+import {
+  CONTRACT_ARCHIVE_LOCAL_VALIDITY_DURATION_MS,
+  getStoredContractArchiveId,
+  storeContractArchiveId,
+  unstoreContractArchiveId,
+} from 'pages/TeacherDashboardContractsLayout/hooks/useDownloadContractArchive/contractArchiveLocalStorage';
+import BulkDownloadContractButton from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockRichieContextFactory({
+    authentication: { backend: 'fonzie', endpoint: 'https://auth.test' },
+    joanie_backend: { endpoint: 'https://joanie.test' },
+  }).one(),
+}));
+
+let mockHasContractToDownload: boolean;
+jest.mock('pages/TeacherDashboardContractsLayout/hooks/useHasContractToDownload/index.tsx', () => ({
+  __esModule: true,
+  default: () => mockHasContractToDownload,
+}));
+
+const mockCheckArchive = jest.fn();
+const mockCreateArchive = jest.fn();
+const mockGetArchive = jest.fn();
+jest.mock('hooks/useContractArchive', () => ({
+  __esModule: true,
+  default: () => ({
+    methods: { get: mockGetArchive, create: mockCreateArchive, check: mockCheckArchive },
+  }),
+}));
+
+describe('TeacherDashboardContractsLayout/BulkDownloadContractButton with fake timer', () => {
+  const Wrapper = ({ children }: PropsWithChildren) => {
+    return (
+      <IntlProvider locale="en">
+        <QueryClientProvider client={createTestQueryClient({ user: true })}>
+          <JoanieApiProvider>{children}</JoanieApiProvider>
+        </QueryClientProvider>
+      </IntlProvider>
+    );
+  };
+
+  let contractArchiveId: string;
+  beforeEach(() => {
+    // organizationId = faker.string.uuid();
+    mockHasContractToDownload = true;
+
+    const now = Date.now();
+    const unvalidCreationTime = now - CONTRACT_ARCHIVE_LOCAL_VALIDITY_DURATION_MS * 2;
+
+    jest.useFakeTimers();
+    jest.setSystemTime(new Date(unvalidCreationTime));
+    contractArchiveId = faker.string.uuid();
+    storeContractArchiveId(contractArchiveId);
+    jest.setSystemTime(new Date(now));
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
```

Comment:
To prevent to issues, it is recommended to exec `jest.runOnlyPendingTimers()` before switching back to real timers.